### PR TITLE
User can sign up, log in and log out

### PR DESCRIPTION
**Pull request by GergKllai1**
Creates sign up and login/logout feature

- Adds temporary welcome controller with index action
- Adds navbar partial for sign up, login, logout links, and a welcome message
- Implements devise gem
- Implements Warden helper
- Adds notices and alerts on the navbar partial and login view
- Sets up haml gem, refactors layout to haml
- Fixes ci.rake formating

PT:
https://www.pivotaltracker.com/n/projects/2229541

### CODE REVIEW

File: app/views/partials/_nav.html.haml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+= notice || alert
+=link_to "Registration" , new_user_registration_path
+=link_to "Login", new_user_session_path
```
Comment (oliverochman) [inline app/views/partials/_nav.html.haml]: put this in the IF statement

File: config/initializers/devise.rb [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+  # ==> Controller configuration
+  # Configure the parent class to the devise controllers.
+  # config.parent_controller = 'DeviseController'
+
+  # ==> Mailer Configuration
+  # Configure the e-mail address which will be shown in Devise::Mailer,
+  # note that it will be overwritten if you use your own mailer class
+  # with default "from" parameter.
+  config.mailer_sender = 'please-change-me-at-config-initializers-devise@example.com'
+
```
Comment (oliverochman) [inline config/initializers/devise.rb]: clean up

File: coverage/.last_run.json [hunk 3]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (oliverochman) [inline coverage/.last_run.json]: add `/coverage/*` to the .gitignore

File: app/views/layouts/application.html.haml [hunk 4]
```diff
@@ -0,0 +1,11 @@
+!!!
+%html
+  %head
+    %meta{content: 'text/html; charset=UTF-8', http: {equiv: "Content-Type"}}/
+    %title CARDnD
+    = csrf_meta_tags
+    = stylesheet_link_tag    'application', media: 'all', data: {'turbolinks-track' => 'reload'}
+    = javascript_include_tag 'application', data: {'turbolinks-track' => 'reload'}
+    %link{href: "https://fonts.googleapis.com/css?family=Roboto", rel: "stylesheet"}/
```
Comment (oliverochman) [inline app/views/layouts/application.html.haml]: remove this font

**Review by oliverochman — CHANGES_REQUESTED** [review]

File: config/initializers/devise.rb (hunk 2 above)
Comment (GergKllai1) [reply config/initializers/devise.rb]: done

File: app/views/partials/_nav.html.haml (hunk 1 above)
Comment (GergKllai1) [reply app/views/partials/_nav.html.haml]: done

File: coverage/.last_run.json (hunk 3 above)
Comment (GergKllai1) [reply coverage/.last_run.json]: done

File: app/views/layouts/application.html.haml (hunk 4 above)
Comment (GergKllai1) [reply app/views/layouts/application.html.haml]: done

**Pull request merged**